add com.janeluo.ikkanalyzer dependency to core model

KeeProMise opened the pull request:
When I run the code in the example to learn the database, I find that the class org/wltea/analyzer/core/IKSegmenter cannot be found. The scope of this class in the module core is provided, and the module example depends on the module core.
![2581683535433_ pic](https://user-images.githubusercontent.com/38941777/236780526-b9bcdeac-0669-4cb3-8d4f-fdba1d489f2f.jpg)
![2591683535489_ pic](https://user-images.githubusercontent.com/38941777/236780597-81d8f833-831a-460a-b122-1cd0418d6d74.jpg)

imbajin commented:
Yep, because it's under the [GPL](https://github.com/blueshen/ik-analyzer/blob/master/LICENSE) license which is not compatible with `Apache License 2.0`, so we have to set the scope to 'provided' and let users to download & config it by themselves

But we do lack the doc to tell the users (so as the mysql dependency - refer https://github.com/apache/incubator-hugegraph/issues/2175#issuecomment-1479049715) , could add some doc & comments in the config [template](https://github.com/apache/incubator-hugegraph/blob/master/hugegraph-dist/src/assembly/static/conf/graphs/hugegraph.properties)

KeeProMise commented:
> Yep, because it's under the [GPL](https://github.com/blueshen/ik-analyzer/blob/master/LICENSE) license which is not compatible with `Apache License 2.0`, so we have to set the scope to `provided` and let users to download & config it by themselves 😿
> 
> But we do lack the doc to tell the users (so as the mysql dependency - refer [#2175 (comment)](https://github.com/apache/incubator-hugegraph/issues/2175#issuecomment-1479049715)) , could add some doc & comments in the config [template](https://github.com/apache/incubator-hugegraph/blob/master/hugegraph-dist/src/assembly/static/conf/graphs/hugegraph.properties) together
> 
> And maybe the **example** should also remove the dependency? @zyxxoo @javeme (also we should enhance the doc for `example` for new users/devs to start

Is it possible to provide a readme in the module example, explaining the dependencies

imbajin commented:
>  Is it possible to provide a readme in the module example, explaining the dependencies

Yep, should add the doc/comments in these spaces:
1. refer a **doc link** in the `README.md` for users to know this ⚠️ 
2. add the details in the doc repo to tell users how to config the dependence (linke ikka/mysql)
3. add the comment in config template to remind users to view **doc link** configuration (could also add the comment in the relative code 👨🏻‍💻)

liuxiaocs7 commented:
Same question, hi, @imbajin

> Yep, because it's under the [GPL](https://github.com/blueshen/ik-analyzer/blob/master/LICENSE?rgh-link-date=2023-05-08T09%3A20%3A06Z) license which is not compatible with Apache License 2.0, so we have to set the scope to provided and let users to download & config it by themselves 😿

The ik tokenizer we used seems to be this one: https://mvnrepository.com/artifact/com.janeluo/ikanalyzer/2012_u6 rather than https://github.com/blueshen/ik-analyzer

```
<dependency>
    <groupId>com.janeluo</groupId>
    <artifactId>ikanalyzer</artifactId>
    <version>${ikanalyzer.version}</version>
</dependency>
```

The license from mvn website, does it meet the requirements?

![image](https://github.com/apache/incubator-hugegraph/assets/42756849/dfffe50d-d509-4ba5-8038-7a6469257d89)

Also [this one](https://github.com/yyn0210/IKAnalyzer) works fine by install local repository.

imbajin commented:
> Same question, hi, @imbajin
> 
> > Yep, because it's under the [GPL](https://github.com/blueshen/ik-analyzer/blob/master/LICENSE?rgh-link-date=2023-05-30T12%3A39%3A39Z) license which is not compatible with Apache License 2.0, so we have to set the scope to provided and let users to download & config it by themselves 😿
> 
> The ik tokenizer we used seems to be this one: [mvnrepository.com/artifact/com.janeluo/ikanalyzer/2012_u6](https://mvnrepository.com/artifact/com.janeluo/ikanalyzer/2012_u6) rather than [blueshen/ik-analyzer](https://github.com/blueshen/ik-analyzer)
> 
> https://github.com/apache/incubator-hugegraph/blob/f23c648937c8c4601a9f721d688eabfec7a8cfb3/hugegraph-core/pom.xml#L198-L203
> 
> The license from mvn website, does it meet the requirements?
> 
> ![image](https://user-images.githubusercontent.com/42756849/241968821-dfffe50d-d509-4ba5-8038-7a6469257d89.png)
> 
> Also [this one](https://github.com/yyn0210/IKAnalyzer) works fine by installing in local repository.

Thanks for the reminder

@javeme @zyxxoo Is it possible to adopt other options?  Consider replacing it

javeme commented:
I think the ikanalyzer license is Apache compatible, but not sure the reason we removed ikanalyzer from https://github.com/apache/incubator-hugegraph/pull/2100

zyxxoo commented:
OK, It seems that we can now use this dependency.

KeeProMise commented:
> OK, It seems that we can now use this dependency. @KeeProMise @liuxiaocs7

I think that since ikanalyzer can be used in the core module, it can also be used in the example module.

imbajin commented:
> > OK, It seems that we can now use this dependency. @KeeProMise @liuxiaocs7
> 
> I think that since ikanalyzer can be used in the core module, it can also be used in the example module.

yep, just fix the dependence check (refer [here](https://github.com/apache/incubator-hugegraph-toolchain/pull/457#issuecomment-1541192642), then we could merge it back)

<img width="643" alt="image" src="https://github.com/apache/incubator-hugegraph/assets/17706099/85e1f66e-0708-4085-99ed-7a5ef96f4df6">


And we could also add the dependence doc in the PR template or Wiki? (for others to know it)

KeeProMise commented:
com.janeluo.ikkanalyze、lucene-queries-4.7.2.jar、lucene-queryparser-4.7.2.jar 、lucene-sandbox-4.7.2.jar is **Apache 2.0**, so I added the above-mentioned third-party dependencies in the known-dependencies.txt file, and added com.janeluo.ikkanalyze to the core module

imbajin reviewed (commented):
Thanks for the contribution, and if we add a new dependency in project, we need also update the info in `LICENSE` & `NOTICE` file both in the `root` & `dist` modules (depend on `source` or `binary` dependency)

@simon824 remove `provided` option, shall we add/update info in dist ([LICENSE](https://github.com/apache/incubator-hugegraph/blob/master/hugegraph-dist/release-docs/LICENSE) & [NOTICE](https://github.com/apache/incubator-hugegraph/blob/master/hugegraph-dist/release-docs/NOTICE)) file (seems yes?)
And we need a **doc** to help new devs know how to add/update the info

simon824 commented:
Hi @KeeProMise ,
If we want to add a new third-party dependency we need to do the following things:
1. Find the third-party dependency repository, and put the dependency license file in [./hugegraph-dist/release-docs/licenses/](https://github.com/apache/incubator-hugegraph/tree/master/hugegraph -dist/release-docs/licenses) path.
2. Declare the dependency in [./hugegraph-dist/release-docs/LICENSE](https://github.com/apache/incubator-hugegraph/blob/master/hugegraph-dist/release-docs/LICENSE) LICENSE information.
3. Find the NOTICE file in the repository and append it to [./hugegraph-dist/release-docs/NOTICE](https://github.com/apache/incubator-hugegraph/blob/master/hugegraph-dist/release-docs /NOTICE) (skip this step if there is no NOTICE file).
4. Execute script [./hugegraph-dist/scripts/dependency/regenerate_known_dependencies.sh](https://github.com/apache/incubator-hugegraph/blob/master/hugegraph-dist/scripts/dependency/regenerate_known_dependencies.sh) to update dependency list [known-dependencies.txt](https://github.com/apache/incubator-hugegraph/blob/master/hugegraph-dist/scripts/dependency/known-dependencies.txt) (or manually update).

KeeProMise commented:
The third-party dependency does not have a NOTICE file, so I took these three steps:

1. Find the third party dependency repository, and put the dependency license file in ./huggegraph-dist/release docs/licenses/ path

2. Declare the dependency in /Hugegraph-dist/release-docs/License License information

3. Execute script /Hugegraph-dist/scripts/dependency/regenerate_ Known_ Dependencies. sh to update dependency list knowledge dependencies. txt

imbajin commented on hugegraph-dist/release-docs/licenses/LICENSE-ikanalyzer-2012_u6.txt:
@simon824 can we ignore the `Apache 2.0` license file?

simon824 replied:
Not sure, I'll check it

javeme reviewed (approved):
LGTM
